crates/sui-framework/packages/sui-system/tests/staking_pool.move

```diff
@@ -247,6 +318,88 @@ module sui_system::staking_pool_tests {
+        let latest_exchange_rate = system_state.pool_exchange_rates(&staking_pool_id).borrow(1);
+        assert!(latest_exchange_rate.sui_amount() == 200 * 1_000_000_000, 0);
+        assert!(latest_exchange_rate.pool_token_amount() == 200 * 1_000_000_000, 0);
+
+        let staked_sui_2 = system_state.request_add_stake_non_entry(
+            coin::mint_for_testing(100 * 1_000_000_000, scenario.ctx()), 
+            @0x1, 
+            scenario.ctx()
+        );
+        test_scenario::return_shared(system_state);
+
+        advance_epoch_with_reward_amounts(0, 200, &mut scenario);
+
+        let mut system_state = scenario.take_shared<SuiSystemState>();
+        let latest_exchange_rate = system_state.pool_exchange_rates(&staking_pool_id).borrow(2);
+        assert!(latest_exchange_rate.sui_amount() == 500 * 1_000_000_000, 0);
+        assert!(latest_exchange_rate.pool_token_amount() == 250 * 1_000_000_000, 0);
+
+        let fungible_staked_sui_1 = system_state.convert_to_fungible_staked_sui(staked_sui_1, scenario.ctx());
+        assert!(fungible_staked_sui_1.value() == 100 * 1_000_000_000, 0);
+        assert!(fungible_staked_sui_1.pool_id() == staking_pool_id, 0);
+
+        let validator = system_state.active_validator_by_address(@0x1);
+        let fungible_staked_sui_data = validator.get_staking_pool_ref().fungible_staked_sui_data();
+        assert!(fungible_staked_sui_data.total_supply() == 100 * 1_000_000_000, 0);
+        assert!(fungible_staked_sui_data.principal_value() == 100 * 1_000_000_000, 0);
+
+        let fungible_staked_sui_2 = system_state.convert_to_fungible_staked_sui(staked_sui_2, scenario.ctx());
+        assert!(fungible_staked_sui_2.value() == 50 * 1_000_000_000, 0);
+        assert!(fungible_staked_sui_2.pool_id() == staking_pool_id, 0);
+
+        let validator = system_state.active_validator_by_address(@0x1);
+        let fungible_staked_sui_data = validator.get_staking_pool_ref().fungible_staked_sui_data();
+        assert!(fungible_staked_sui_data.total_supply() == 150 * 1_000_000_000, 0);
+        assert!(fungible_staked_sui_data.principal_value() == 200 * 1_000_000_000, 0);
+        test_scenario::return_shared(system_state);
+
+        // test starts here
+        advance_epoch_with_reward_amounts(0, 500, &mut scenario);
+
+        let mut system_state = scenario.take_shared<SuiSystemState>();
+        let latest_exchange_rate = system_state.pool_exchange_rates(&staking_pool_id).borrow(scenario.ctx().epoch());
+        std::debug::print(latest_exchange_rate);
+        assert!(latest_exchange_rate.sui_amount() == 1000 * 1_000_000_000, 0);
+        assert!(latest_exchange_rate.pool_token_amount() == 250 * 1_000_000_000, 0);
+
+        let sui_1 = system_state.redeem_fungible_staked_sui(fungible_staked_sui_1, scenario.ctx());
+        assert!(sui_1.value() == 400 * 1_000_000_000 - 1, 0);
+        test_scenario::return_shared(system_state);
+
+        advance_epoch_with_reward_amounts(0, 0, &mut scenario);
```

Comment:
can we advance with some compute rewards in either this line or line 396? just trying to increase the scenario coverage